src/averageAges.js

```diff
@@ -1,61 +1,53 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let manArray = [];
+
+  if (century) {
+    manArray = people.filter((a) => a.sex === 'm'
+    && Math.ceil(a.died / 100) === century);
+  } else {
+    manArray = people.filter((a) => a.sex === 'm');
+  }
+
+  const manAges = manArray.map((b) => b.died - b.born);
+
+  return manAges.reduce((a, b) => (a + b)) / manAges.length;
```

Comment:
you can combine these methods.

```suggestion
  return manArray.reduce(
    (averageAge, person) => averageAge + (person.died - person.born),
    0,
  ) / manAges.length;
```

`- 1 loop )`

Reply:
Also, you do the same on calculateWomenAverageAge. You can create separate function and you it here and on calculateWomenAverageAge